lib/policy_machine.rb

```diff
@@ -244,6 +244,14 @@ def batch_find(type:, query: {}, config: {}, &blk)
     end
   end
 
+  def pluck(type:, fields:, options:)
+    if policy_machine_storage_adapter.respond_to?(:pluck)
+      policy_machine_storage_adapter.send(:pluck, { type: type, fields: fields, options: options })
+    else
+      throw MethodNotSupportedError.new("Method 'pluck' is not implemented for storage adapter #{policy_machine_storage_adapter.class}")
```

Comment:
`throw` is extremely rare in Ruby. I've only actually seen it used in a handful of codebases. When it's used, iirc it should really only be used with catch. Otherwise we should probably just use `raise`. I think that's the case here. Let's just use `raise`.

Reply:
fixed